pex/bin/pex.py

```diff
@@ -892,6 +892,27 @@ def build_pex(
         "Adding distributions from pexes: {}".format(" ".join(options.requirements_pexes))
     ):
         for requirements_pex in options.requirements_pexes:
+            if (
+                pex_info.deps_are_wheel_files
+                != PexInfo.from_pex(requirements_pex).deps_are_wheel_files
+            ):
+                die(
+                    "The {option} option was selected but the --requirements-pex "
+                    "{requirements_pex} is built with {opposite_option}. Any --requirements-pex "
+                    "you want to merge into the main PEX must be built with {option}.".format(
```

Comment:
Hm, I'm a bit confused, as a test like the following seems to have matching options but still fails with the same "Is a directory" error:

```shell
pex --no-pre-install-wheels -o /tmp/first.pex cowsay
pex --no-pre-install-wheels -o /tmp/second.pex --requirements-pex /tmp/first.pex --layout packed
```

This error message suggests this should work. Am I misunderstanding it?

Reply:
It works in the test in this PR (minus packed, your grid made layout appear irrelevant).

Reply:
Aha, I read poorly. Back to drawing board.

Reply:
@huonw I don't repro:
```
:; ./dtox.sh inspect
jsirois@617430d8f0ac:/development/pex$ python -mpex --no-pre-install-wheels -o /tmp/first.pex cowsay
jsirois@617430d8f0ac:/development/pex$ python -mpex --no-pre-install-wheels -o /tmp/second.pex --requirements-pex /tmp/first.pex --layout packed
jsirois@617430d8f0ac:/development/pex$ PEX_SCRIPT=cowsay /tmp/second.pex/__main__.py -t Moo!
  ____
| Moo! |
  ====
    \
     \
       ^__^
       (oo)\_______
       (__)\       )\/\
           ||----w |
           ||     ||
jsirois@617430d8f0ac:/development/pex$
exit
```

Let me dig a bit more to see if I can gedanken whatever it is you're hitting.

Reply:
Yeah @huonw I've got nothing. The new matrixed test passes locally and in CI as did the by-hand replication above; so I'm not sure what is different about our environments that I'm missing. I think this PR is forward progress though and, afaict, introduces no new bugs, just whittling down the existing buggy cases at worst; so I'll proceed. If you can find maybe a dockerable repro case (I used `./dtox.sh inspect` above for this reason - so we can share ~the same env) then I can address in a follow up.

Reply:
Aha, ok - Mac specific issue: https://github.com/pex-tool/pex/actions/runs/8457687342/job/23170224480?pr=2392#step:9:2348

Pain in the butt, but at least I have a repro.

Reply:
Ah, sorry, I didn't cross my mind that it might be platform-specific. I'll have some time over the next few days so can help fix if doing loops through CI is too unpleasant.

Reply:
Ok, the issue is systemic to `--requirements-pex X --no-pre-install-wheels`; it's just masked on Linux. I'll add a fix patch to this PR tomorrow.